lib/active_merchant/billing/gateways/visanet_peru.rb

```diff
@@ -197,15 +200,24 @@ def success_from(response)
       end
 
       def message_from(response, options, action)
-        if empty?(response['errorMessage']) || response['errorMessage'] == '[ ]'
-          action == 'refund' ? "#{response['data']['DSC_COD_ACCION']}, #{options[:error_message]}" : response['data']['DSC_COD_ACCION']
-        elsif action == 'refund'
-          message = "#{response['errorMessage']}, #{options[:error_message]}"
-          options[:error_message] = response['errorMessage']
-          message
-        else
-          response['errorMessage']
-        end
+        message_from_messages(
+          response['errorMessage'],
+          action_code_description(response),
+          options[:error_message]
+        )
+      end
+
+      def message_from_messages(*args)
+        args.reject { |m| error_message_empty?(m) }.join(' | ')
+      end
+
+      def action_code_description(response)
+        return nil unless response['data']
+        response['data']['DSC_COD_ACCION']
+      end
+
+      def error_message_empty?(error_message)
+        empty?(error_message) || error_message == '[ ]'
```

Comment:
Could use `.blank?`

Reply:
`empty?` was already in use here (I just extracted to a method to clarify the conditionals elsewhere) so I retained it for lack of a reason to change it. Is `.blank?` to be preferred to `empty?` here?

Reply:
Not a big deal; in fact there's a vague pressure to avoid dependencies on ActiveSupport.